### Урок 6. Сравнение долей. Построение доверительного интервала



In [1]:
import numpy as np

#### Задача 1.

Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.

Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, 

а объем выборки n = 256.

n = 256

M = 80

a = 0.95

$\sigma$ = 16

$$X_{ср}+/-Z_{\frac{\alpha}{2}}*\frac{\sigma}{\sqrt{n}}$$

In [2]:
80 + 1.96 *(16/np.sqrt(256))

81.96

In [3]:
80 - 1.96 *(16/np.sqrt(256))

78.04

Доверительный интервал для оценки математического ожилания данной выборки принадлежит [78.04; 81.96]

#### Задача 2.

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:

6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1

Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,

оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [8]:
import numpy as np
X = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
X

[6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]


$$X_{ср}+/-t_{\frac{\alpha}{2}}*\frac{\sigma}{\sqrt{n}}$$


In [11]:
sigma = np.std(X, ddof=1)
sigma

0.4508017549014448

In [12]:
len(X)

10

$t_{\frac{\alpha}{2}} = 2.262 $

In [15]:
X_avg = np.mean(X)
X_avg

6.590000000000001

In [17]:
6.59-2.262*(sigma/np.sqrt(10))

6.2675382559124255

In [18]:
6.59+2.262*(sigma/np.sqrt(10))

6.912461744087574

С вероятностью 95% истинное значение величины Х будет принадлежать интервалу [6.27; 6.91]

#### Задача 3.

На препарате А положительный результат лечения наблюдается у 17 из 32 пациентов, а на препарате В у 9 из 22. 

Построить 95% доверительный интервал для разности долей.

Обнаружены ли статистически значимые различия?

n_a = 32 
m_a = 7 
p_a = 7/32 = 0.22

n_b = 22
m_b = 9
p_b = 9/22 = 0.41

In [21]:
p = (7+9)/(32+22)
p


0.2962962962962963

In [22]:
z = ((7/32 - 9/22) - 0.5*(1/32 + 1/22))/np.sqrt(p*(1-p)*(1/32 + 1/22))
z

-1.8083589367045831

z_t = +/- 1.96

Рассчётное значение z попадает в интервал, значит статистически значимых различий нет.